odoo/addons/test_lint/tests/test_docstring.py

```diff
@@ -258,6 +259,12 @@ def _test_docstring_params(self, method, doctree):
         for param, doc_type in doc_types.items():
             sign_type = sign_types.get(param, signature.empty)
             if sign_type != signature.empty:
-                if isinstance(sign_type, type):
-                    sign_type = sign_type.__name__
+                sign_type = self._stringify_annotation(sign_type)
                 self.assertEqual(sign_type, doc_type)
+
+    def _stringify_annotation(self, sign_type):
+        if isinstance(sign_type, type):
+            sign_type = sign_type.__name__
+        elif not isinstance(sign_type, str):
+            sign_type = str(sign_type)
+        return sign_type
```

Comment:
IMO this will be faster
```suggestion
        return str(sign_type)
```